superset/db_engine_specs/base.py

```diff
@@ -1734,9 +1734,10 @@ def validate_database_uri(cls, sqlalchemy_uri: URL) -> None:
 
         :param sqlalchemy_uri:
         """
-        for query_param in sqlalchemy_uri.query.keys():
-            if query_param in cls.disallow_uri_query_params:
-                raise ValueError("Disallowed query parameter")
+        if existing_disallowed := cls.disallow_uri_query_params.intersection(
+            sqlalchemy_uri.query
+        ):
+            raise ValueError("Disallowed query parameter(s) %s", existing_disallowed)
```

Comment:
nit: I believe we use "forbidden" in a few other places:
```suggestion
            raise ValueError("Forbidden query parameter(s): %s", existing_disallowed)
```